src/libraries/System.Private.Xml/src/System/Xml/Serialization/Mappings.cs

```diff
@@ -981,6 +981,31 @@ internal MemberMapping Clone()
         {
             return new MemberMapping(this);
         }
+
+        internal bool Hides(MemberMapping mapping)
+        {
+            // Semantics. But all mappings would hide a null mapping, so indicating that this particular instance of mapping
+            // specifically hides a null mapping is not necessary and could be misleading. So just return false.
+            if (mapping == null)
+                return false;
+
+            var baseMember = mapping.MemberInfo;
+            var derivedMember = this.MemberInfo;
+
+            // Similarly, if neither mapping has any MemberInfo, then its not appropriate to say that one hides the other.
+            if (baseMember == null || derivedMember == null)
+                return false;
+
+            // If the member names are different, they don't hide each other
+            if (baseMember.Name != derivedMember.Name)
+                return false;
```

Comment:
If the prior comparison does become &&, then null coalescing will be needed to be used here as one of them could be null.